### PR TITLE
Track first pageview in a session

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Changes

PRs over issues.

This tracks the first pageview in a session so I can see where my users landed. 

Could also be set as a super prop like we do with referrers?

User request.

## Checklist
- [ ] Tests for new code (if applicable)
- [ ] TypeScript definitions (module.d.ts) updated and in sync with library exports (if applicable)

### CODE REVIEW

File: src/posthog-core.js [hunk 1]
```diff
@@ -629,7 +629,14 @@ PostHogLib.prototype.capture_pageview = function (page) {
     if (_.isUndefined(page)) {
         page = document.location.href
     }
-    this.capture('$pageview')
+    const isFirstPageviewInSession = !window.sessionStorage.getItem('captured-first-pageview')
```
Comment (macobo) [inline src/posthog-core.js:632]: We can't assume sessionStorage is available or if it can be used.

**Review by macobo — COMMENTED** [review]
Should this be part of posthog core/is it useful enough? It will increase all users data amount by a significant amount.

Perhaps a better way would be to expose hooks like this?

```
posthog.init('tolen', {
  onCapture: ($event, payload) => { ... }
})
```

and implement this via that?

**Comment by yakkomajuri** [comment]
That could be a good approach, yes. As for being part of core, tbh I kinda think so. It's something people migrating from GA will miss, and I was surprised we didn't actually have a way of knowing this

**Comment by macobo** [comment]
Not clear from this; What is the insight user is after/what query are they trying to do?

**Comment by yakkomajuri** [comment]
They want to see a ranking of the pages where the user first landed on their website for a session

**Comment by yakkomajuri** [comment]
So with this I could do:

`$pageview where $is_first_pageview = true, breakdown by pathname`

**Pull request closed without merging**